climakitae/threshold_tools.py

```diff
@@ -884,3 +889,523 @@ def return_period(ams):
+        window_size = dur_len 
+
+        # The "min" operation will return 0 if any time in the window is not an
+        # event, which is the behavior we want. It will only return 1 for True 
+        # if all values in the duration window are 1.
+        events_da = events_da.rolling(time = window_size, center=False).min("time")
+
+    # Groupby 
+    if groupby is not None:
+        if (groupby == (1, "hour") and da.frequency == "1hr") \
+            or (groupby == (1, "day") and da.frequency == "1day") \
+            or (groupby == (1, "month") and da.frequency == "1month") \
+            or groupby == duration1:
+            # groupby specification is the same as data frequency, do nothing
+            pass
+        else:
+            group_len, group_type = groupby
+            indexer_type = str.capitalize(group_type[0]) # capitalize the first letter to use as the indexer (i.e. H, D, M, or Y)
+            group_totals = events_da.resample(time=f"{group_len}{indexer_type}", label="left").sum() # sum occurences within each group
+            events_da = (group_totals > 0).where(group_totals.isnull()==False) # turn back into a boolean with preserved NaNs (0 or 1 for whether there is any occurance in the group)
+
+    return events_da
+
+def _exceedance_count_name(exceedance_count):
+    """
+    Helper function to build the appropriate name for the queried exceedance count.
+    Examples:
+        'Number of hours'
+        'Number of days'
+        'Number of 3-day events'
+    """
+    # If duration is used, this determines the event name
+    dur = exceedance_count.duration2
+    if dur is not None:
+        d_num, d_type = dur 
+        if d_num != 1:
+            event = f"{d_num}-{d_type} events"
+        else:
+            event = f"{d_type}s" # ex: day --> days
+    else:
+        # otherwise use "groupby" if not None
+        grp = exceedance_count.group
+        if grp is not None:
+            g_num, g_type = grp
+            if g_num != 1:
+                event = f"{g_num}-{g_type} events"
+            else:
+                event = f"{g_type}s" # ex: day --> days
+        else:
+            # otherwise use data frequency info as the default event type
+            if exceedance_count.frequency == "1hr":
+                event = "hours"
+            elif exceedance_count.frequency == "1day":
+                event = "days"
+            elif exceedance_count.frequency == "1month":
+                event = "months"
+
+    return f"Number of {event}"
+
+def plot_exceedance_count(exceedance_count):
+    """
+    Plots each simulation as a different color line.
+    Drop down option to select different scenario.
+    Currently can only plot for one location, so is expecting input to already be subsetted or an area average.
+    """
+    plot_obj = exceedance_count.hvplot.line(
+        x="time", 
+        widget_location="bottom", 
+        by="simulation", 
+        groupby=["scenario"],
```

Comment:
I don't think I like this scenario dropdown, because I would expect that there's only one scenario option always-- right? Is that a correct assumption? If so, I would make that dropdown disappear, and add the scenario information as a subtitle to the plot. If that's not a correct assumption, and you'd expect that with different data selections (or in future iterations of this panel), there would be more than one scenario option, definitely leave the dropdown in.  

<img width="600" alt="only one dropdown option" src="https://user-images.githubusercontent.com/66140951/192859537-bdf3c4f7-d960-4a1a-ae11-20bf21e034de.png">